ui/app/AppLayouts/Wallet/stores/ActivityFiltersStore.qml

```diff
@@ -70,6 +70,32 @@ QtObject {
             root.toTimestamp = 0
             toTimestampNoLimit = true
         }
+
+        function getCollectiblesFilterJson() {
+            let collectibles = []
+            for(let i = 0 ; i < collectiblesFilter.length ; i++) {
+                const uid = collectiblesFilter[i]
+
+                // Using cache becauses collectibles model is not fully fetched when switching accounts
+                if (uid in collectiblesDataCache) {
+                    collectibles.push(collectiblesDataCache[uid])
+                    continue;
+                }
+
+                let obj = {}
+                const tokenId = Number(collectiblesList.getTokenId(uid)).toString(16)
+                // NOTE passing only '0x' will raise parse error in backend
```

Comment:
Is there a more straightforward way? Having concerns about the backend so high up in the presentation layer makes me think there may be a better way.
A usual approach I employ with models is to refer to items by their index instead of identity. What do you think about this approach? Is it feasible and simplifies things instead of token identity?

Reply:
If we have to keep the selection between account switching, it makes sense to keep using the identity.